### PR TITLE
whitebit: stop orders (clean)

### PR DESCRIPTION
**Pull request by DoctorSlimm**

### CODE REVIEW

File: js/whitebit.js [hunk 1]
```diff
@@ -974,64 +974,65 @@ module.exports = class whitebit extends Exchange {
...
+                    if (price === undefined) {
+                        throw new InvalidOrder (this.id + " createOrder () requires the price argument with market buy orders to calculate total order cost (amount to spend), where cost = amount * price. Supply a price argument to createOrder() call if you want the cost to be calculated for you from price and amount, or, alternatively, add .options['createMarketBuyOrderRequiresPrice'] = false and supply the total cost value in the 'amount' argument");
                     }
+                    cost = amount * price;
                 } else {
-                    cost = (cost === undefined) ? amount : cost;
+                    cost = amount;
                 }
-                request['amount'] = this.costToPrecision (symbol, cost);
+                request['amount'] = this.priceToPrecision (symbol, cost);
```
Comment (carlosmiei) [inline js/whitebit.js:1032]: @DoctorSlimm, why are we using `priceToPrecision` instead of `costToPrecision` here?

Comment (frosty00) [reply js/whitebit.js:1032]: actually that looks like it should be `amountToPrecision` )

Comment (carlosmiei) [reply js/whitebit.js:1032]: @frosty00 nop because this is a `createMarketBuyOrderRequiresPrice` case 🙂

**Comment by carlosmiei** [comment]
@DoctorSlimm all good now, thanks ✅ ✅ 

```
p whitebit createOrder "LTC/USDT" "market" "buy" 0.12 50 '{"stopPrice": 30}'
Python v3.9.7
CCXT v1.91.60
whitebit.createOrder(LTC/USDT,market,buy,0.12,50,{'stopPrice': 30})
{'amount': None,
 'average': None,
 'clientOrderId': None,
 'cost': 5.99,
 'datetime': '2022-08-03T17:29:05.016Z',
 'fee': {'cost': 0.0, 'currency': 'USDT'},
 'fees': [{'cost': 0.0, 'currency': 'USDT'}],
 'filled': 0.0,
 'id': '107181508343',
 'info': {'activation_price': '30',
          'amount': '5.99',
          'clientOrderId': '',
          'dealFee': '0',
          'dealMoney': '0',
          'dealStock': '0',
          'left': '5.99',
          'makerFee': '0',
          'market': 'LTC_USDT',
          'orderId': '107181508343',
          'side': 'buy',
          'takerFee': '0.001',
          'timestamp': '1659547745.016335',
          'type': 'stop market'},
 'lastTradeTimestamp': None,
 'postOnly': None,
 'price': None,
 'remaining': None,
 'side': 'buy',
 'status': None,
 'stopPrice': 30.0,
 'symbol': 'LTC/USDT',
 'timeInForce': 'IOC',
 'timestamp': 1659547745016,
 'trades': [],
 'type': 'market'}
 
 ```

```
p whitebit createOrder "LTC/USDT" "market" "buy" 0.15 50 '{"timeInForce":"FOK"}'
Python v3.9.7
CCXT v1.91.60
whitebit.createOrder(LTC/USDT,market,buy,0.15,50,{'timeInForce': 'FOK'})
{'amount': 0.15,
 'average': None,
 'clientOrderId': None,
 'cost': None,
 'datetime': '2022-08-03T17:16:58.853Z',
 'fee': {'cost': 0.0089385, 'currency': 'USDT'},
 'fees': [{'cost': 0.0089385, 'currency': 'USDT'}],
 'filled': 0.15,
 'id': '107178985347',
 'info': {'amount': '0.15',
          'clientOrderId': '',
          'dealFee': '0.0089385',
          'dealMoney': '8.9385',
          'dealStock': '0.15',
          'left': '0',
          'makerFee': '0',
          'market': 'LTC_USDT',
          'orderId': '107178985347',
          'side': 'buy',
          'takerFee': '0.001',
          'timestamp': '1659547018.853589',
          'type': 'stock market'},
 'lastTradeTimestamp': None,
 'postOnly': None,
 'price': None,
 'remaining': 0.0,
 'side': 'buy',
 'status': None,
 'stopPrice': None,
 'symbol': 'LTC/USDT',
 'timeInForce': 'FOK',
 'timestamp': 1659547018853,
 'trades': [],
 'type': 'market'}
 ```

```
 p whitebit createOrder "LTC/USDT" "limit" "buy" 0.2 40
Python v3.9.7
CCXT v1.91.60
whitebit.createOrder(LTC/USDT,limit,buy,0.2,40)
{'amount': 0.2,
 'average': None,
 'clientOrderId': None,
 'cost': 0.0,
 'datetime': '2022-08-03T17:17:20.433Z',
 'fee': {'cost': 0.0, 'currency': 'USDT'},
 'fees': [{'cost': 0.0, 'currency': 'USDT'}],
 'filled': 0.0,
 'id': '107179052649',
 'info': {'amount': '0.2',
          'clientOrderId': '',
          'dealFee': '0',
          'dealMoney': '0',
          'dealStock': '0',
          'left': '0.2',
          'makerFee': '0.001',
          'market': 'LTC_USDT',
          'orderId': '107179052649',
          'price': '40',
          'side': 'buy',
          'takerFee': '0.001',
          'timestamp': '1659547040.433655',
          'type': 'limit'},
 'lastTradeTimestamp': None,
 'postOnly': None,
 'price': 40.0,
 'remaining': 0.2,
 'side': 'buy',
 'status': None,
 'stopPrice': None,
 'symbol': 'LTC/USDT',
 'timeInForce': None,
 'timestamp': 1659547040433,
 'trades': [],
 'type': 'limit'}
 ```

```
 n whitebit createOrder "LTC/USDT" "market" "buy" 0.2 50           
Debugger attached.
2022-08-03T17:26:38.334Z
Node.js: v18.4.0
CCXT v1.91.60
(node:5202) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
whitebit.createOrder (LTC/USDT, market, buy, 0.2, 50)
2022-08-03T17:26:40.125Z iteration 0 passed in 612 ms

{
  info: {
    orderId: '107181059489',
    clientOrderId: '',
    market: 'LTC_USDT',
    side: 'buy',
    type: 'market',
    timestamp: '1659547600.125503',
    dealMoney: '9.9896025',
    dealStock: '0.16719',
    amount: '9.99',
    takerFee: '0.001',
    makerFee: '0',
    left: '0.0003975',
    dealFee: '0.0099896025'
  },
  id: '107181059489',
  symbol: 'LTC/USDT',
  clientOrderId: undefined,
  timestamp: 1659547600125,
  datetime: '2022-08-03T17:26:40.125Z',
  lastTradeTimestamp: undefined,
  timeInForce: 'IOC',
  postOnly: undefined,
  status: undefined,
  side: 'buy',
  price: 59.75237753454154,
  type: 'market',
  stopPrice: undefined,
  amount: undefined,
  filled: 0.16719,
  remaining: undefined,
  average: 59.75237753454154,
  cost: 9.99,
  fee: { cost: 0.0099896025, currency: 'USDT' },
  trades: [],
  fees: [ { cost: 0.0099896025, currency: 'USDT' } ]
}
2022-08-03T17:26:40.125Z iteration 1 passed in 612 ms
````

**Review by carlosmiei — APPROVED** [review]

**Pull request merged**